rotkehlchen/chain/ethereum/tokens.py

```diff
@@ -141,116 +156,59 @@ def query_tokens_for_addresses(
+                token_amounts = self._query_chunks(
+                    account=account,
+                    chunks=get_chunks(all_tokens, n=OTHER_MAX_TOKEN_CHUNK_LENGTH),
+                    querying_etherscan=False,
                 )
-                continue
-            balances[token] += value
-            if token in token_usd_price:
-                continue
-            # else get the price
-            try:
-                usd_price = Inquirer().find_usd_price(token)
-            except RemoteError:
-                usd_price = Price(ZERO)
-            token_usd_price[token] = usd_price
+            else:
+                token_amounts = self._query_chunks(
+                    account=account,
+                    chunks=get_chunks(all_tokens, n=ETHERSCAN_MAX_TOKEN_CHUNK_LENGTH),
+                    querying_etherscan=True,
+                )
+            self.db.save_tokens_for_address(write_cursor, account, list(token_amounts.keys()))
 
-    def _get_multitoken_account_balance(
+    def query_tokens_for_addresses(
             self,
-            tokens: List[EthereumToken],
-            account: ChecksumEthAddress,
-            call_order: Optional[Sequence[WeightedNode]],
-    ) -> Dict[str, FVal]:
-        """Queries balances of multiple tokens for an account
-
-        Return a dictionary with keys being tokens and value a dictionary of
-        account to balances
+            addresses: List[ChecksumEthAddress],
+    ) -> TokensReturn:
+        """Queries token balances for a list of addresses
+        Returns the token balances of each address and the usd prices of the tokens
 
         May raise:
         - RemoteError if an external service such as Etherscan is queried and
           there is a problem with its query.
         - BadFunctionCallOutput if a local node is used and the contract for the
           token has no code. That means the chain is not synced
         """
-        log.debug(
-            'Querying ethereum chain for multi token account balances',
-            eth_address=account,
-            tokens_num=len(tokens),
-        )
-        balances: Dict[str, FVal] = {}
-        result = ETH_SCAN.call(
-            ethereum=self.ethereum,
-            method_name='tokensBalance',
-            arguments=[account, [x.ethereum_address for x in tokens]],
-            call_order=call_order,
-        )
-        for tk_idx, token in enumerate(tokens):
-            token_amount = result[tk_idx]
-            if token_amount != 0:
-                normalized_amount = token_normalized_value(token_amount, token)
-                log.debug(
-                    f'Found {token.symbol}({token.ethereum_address}) token balance for '
-                    f'{account} and amount {normalized_amount}',
-                )
-                balances[token.identifier] = normalized_amount
-        return balances
+        token_usd_price: Dict[EthereumToken, Price] = {}
+        result = {}
```

Comment:
rename for code clarity and readability

```suggestion
        address_to_balance = {}
```

Reply:
Done